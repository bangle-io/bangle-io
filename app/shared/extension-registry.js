import editorCore from 'editor-core/index';
import inlineCommandPalette from 'inline-command-palette/index';
import inlineBacklinkPalette from 'inline-backlink/index';
import collapsibleHeading from 'collapsible-heading/index';
import imageExtension from 'image-extension/index';
import inlineEmoji from 'inline-emoji/index';
import collabExtension from 'collab-extension/index';
import editorScrollExtension from 'editor-scroll-extension';
import corePalette from 'core-palettes';
import coreActions from 'core-actions';
import { ExtensionRegistry } from 'extension-registry/index';

// TODO move this async, i think a promise should be fine.
export const extensionRegistry = new ExtensionRegistry([
  inlineEmoji,
  editorCore,
  collabExtension,
  inlineCommandPalette,
  inlineBacklinkPalette,
  collapsibleHeading,
  imageExtension,
  editorScrollExtension,
  coreActions,
  // NOTE: keep the core palette last
  // as it has note palette in it
  corePalette,
]);
