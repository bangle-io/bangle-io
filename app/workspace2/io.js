import * as idb from 'idb-keyval';

export class FileOps {
  // file
  static async getFile(wsPath) {
    return idb.get(wsPath);
  }

  static async deleteFile(wsPath) {
    const file = await this.getFile(wsPath);
    if (!file) {
      throw new FileError(`File ${wsPath} does not exists`);
    }
    return idb.del(wsPath);
  }

  static async renameFile(wsPath, newWsPath) {
    const file = await this.getFile(wsPath);

    if (!file) {
      throw new FileError(`File ${wsPath} does not exists`);
    }

    const newFile = await this.getFile(newWsPath);

    if (newFile) {
      throw new FileError(`File ${newWsPath} already exists`);
    }

    await this.deleteFile(wsPath);
    return this.createFile(newWsPath, file);
  }

  static async createFile(wsPath, payload) {
    if (await this.getFile(wsPath)) {
      throw new FileError(`File ${wsPath} already exists`);
    }
    return idb.set(wsPath, payload);
  }

  static async updateFile(wsPath, payload) {
    if (!(await this.getFile(wsPath))) {
      throw new FileError(`File ${wsPath} does not exists`);
    }
    return idb.set(wsPath, payload);
  }

  static async listFiles(wsName) {
    const keys = (await idb.keys()) || [];

    return keys.filter((k) => k.startsWith(wsName + ':'));
  }
}

export class FileError extends Error {
  constructor(message) {
    super('FileError: ' + message);
    this.name = 'FileError';
  }
}
export class FilePermissionError extends Error {
  constructor(message) {
    super('FilePermissionError: ' + message);
    this.name = 'FilePermissionError';
  }
}
