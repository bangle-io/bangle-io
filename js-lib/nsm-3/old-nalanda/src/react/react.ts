import { useDebugValue, useRef, useState } from 'react';
import useSyncExternalStoreExports from 'use-sync-external-store/shim';

import type { Slice } from '../vanilla/slice';
import type { Store } from '../vanilla/store';
import type { AnySlice } from '../vanilla/types';

const { useSyncExternalStore } = useSyncExternalStoreExports;

type InferData<T> = T extends Slice<any, infer TData, any> ? TData : never;

interface ReactAdapter<TSnapshot> {
  subscribe: (onStoreChange: () => void) => () => void;
  getSnapshot: () => TSnapshot;
}

export function createUseTrackSliceHook<TAllSliceName extends string = any>(
  store: Store<TAllSliceName>,
) {
  function useSlice<TSlice extends AnySlice>(sl: TSlice): InferData<TSlice> {
    const ref = useRef<InferData<TSlice>>();

    const [adapter] = useState(() => {
      const adapter: ReactAdapter<InferData<TSlice>> = {
        subscribe: (onStoreChange) => {
          const sliceEffect = store.effect((effectStore) => {
            const selectedData = sl.track(effectStore);
            ref.current = selectedData;

            onStoreChange();

            queueMicrotask(() => {
              if (effectStore._runInstance?._addTrackedCount === 0) {
                console.warn(
                  `You forgot to destructure/access the tracked field from ${sl.name}. This will not track any changes!.`,
                );
              }
            });
          });

          return () => {
            store.unregisterEffect(sliceEffect);
          };
        },
        getSnapshot: () => {
          return ref.current ?? sl.get(store.state);
        },
      };

      return adapter;
    });

    const data = useSyncExternalStore(adapter.subscribe, adapter.getSnapshot);

    useDebugValue(data);

    return data;
  }

  return useSlice;
}
