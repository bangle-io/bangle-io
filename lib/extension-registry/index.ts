/// <reference path="../../missing-types.d.ts" />

export * from './Extension';
export * from './ExtensionRegistry';
export * from './ExtensionRegistryContext';
export {
  extensionRegistryEffects,
  nsmExtensionRegistry,
} from './nsm-extension-registry-slice';
export * from './PluginType';
