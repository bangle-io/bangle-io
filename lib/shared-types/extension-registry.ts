import type { SpecRegistry } from '@bangle.dev/core';
import type { Node, PluginKey } from '@bangle.dev/pm';

export type { ExtensionRegistry } from '@bangle.io/extension-registry';
export type DialogComponentType = React.ComponentType<{
  dialogName: string;
  onDismiss: (dialogName: string) => void;
}>;

export interface DialogType {
  name: `dialog::${string}`;
  ReactComponent: DialogComponentType;
}

export type SerialOperationNameType = string;
export interface SerialOperationType {
  name: SerialOperationNameType;
  value?: any;
}
export interface SerialOperationDefinitionType {
  name: SerialOperationType['name'];
  title: string;
  keywords?: string[];
  keybinding?: string;
  // when true, will hide it from the user
  hidden?: boolean;
  // when true, will prevent editor focus when executed
  preventEditorFocusOnExecute?: boolean;
}
export type DispatchSerialOperationType = (
  sOperation: SerialOperationType,
) => void;

export type SerialOperationHandler = (
  sOperation: SerialOperationType,
) => boolean | undefined;

export type SerialOperationKeybindingMapping = Record<
  SerialOperationNameType,
  string
>;

export interface NoteSidebarWidget {
  name: `note-sidebar-widget::${string}`;
  ReactComponent: React.ComponentType;
  title: string;
}

export interface EditorWatchPluginState {
  // The operation to dispatch when state changes
  // Bangle.io will passe an editorId when dispatching this sOperation
  // to help identify which editors plugin state changed.
  operation: SerialOperationNameType;
  // The pluginKey of the plugin's state to watch
  pluginKey: PluginKey;
}

export interface NoteFormatProvider {
  name: string;

  extensions: string[];

  description: string;

  serializeNote: (
    doc: Node,
    specRegistry: SpecRegistry,
    fileName: string,
  ) => string;

  // Return PMNode if parsing is successful, otherwise return undefined
  parseNote: (
    value: string,
    specRegistry: SpecRegistry,
    // only applicable if dealing with prosemirror specific markdown
    markdownPlugins: any[],
  ) => Node | undefined;
}

export type OnStorageProviderError = (error: Error) => boolean;
