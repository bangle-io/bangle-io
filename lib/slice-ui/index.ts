export {
  type UISliceState,
  initialUISliceState as initialState,
} from './constants';
export { nsmUISlice, uiEffects } from './nsm-ui-slice';
export * as nsmUI from './nsm-ui-slice';
