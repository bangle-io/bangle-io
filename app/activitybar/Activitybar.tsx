import React from 'react';

import { useNsmSlice, useNsmSliceState } from '@bangle.io/bangle-store-context';
import { CHANGELOG_MODAL_NAME } from '@bangle.io/constants';
import { vars } from '@bangle.io/css-vars';
import { useExtensionRegistryContext } from '@bangle.io/extension-registry';
import { nsmSliceWorkspace } from '@bangle.io/nsm-slice-workspace';
import { goToWorkspaceHome, nsmPageSlice } from '@bangle.io/slice-page';
import { nsmUI, nsmUISlice } from '@bangle.io/slice-ui';
import { Button, GiftIcon, SingleCharIcon } from '@bangle.io/ui-components';
import { cx } from '@bangle.io/utils';

import { ActivitybarOptionsDropdown } from './ActivitybarOptionsDropdown';
import { ButtonStyleOBj } from './common';

export function Activitybar() {
  const extensionRegistry = useExtensionRegistryContext();
  const operationKeybindings =
    extensionRegistry.getSerialOperationKeybindingMapping();
  const { wsName } = useNsmSliceState(nsmSliceWorkspace);
  const [, pageDispatch] = useNsmSlice(nsmPageSlice);

  const sidebars = extensionRegistry.getSidebars();

  const [{ changelogHasUpdates, sidebar, widescreen }, uiDispatch] =
    useNsmSlice(nsmUISlice);

  const sideBarComponents = sidebars
    .filter((r) => {
      return r.activitybarIconShow ? r.activitybarIconShow(wsName) : true;
    })
    .map((r) => {
      const active = sidebar === r.name;

      return (
        <Button
          ariaLabel={r.hint}
          key={r.name}
          onPress={() => {
            uiDispatch(nsmUI.toggleSideBar(r.name));
          }}
          style={{
            ...ButtonStyleOBj.normal,
            borderColor: active
              ? vars.color.promote.border
              : vars.misc.activitybarBg,
          }}
          onPressStyle={ButtonStyleOBj.press}
          onHoverStyle={ButtonStyleOBj.hover}
          className={cx('border-l-2', active && 'BU_is-active')}
          variant="transparent"
          size="lg"
          leftIcon={r.activitybarIcon}
        />
      );
    });

  return (
    <div
      data-testid="app-activitybar_activitybar"
      style={{
        backgroundColor: vars.misc.activitybarBg,
        color: vars.misc.activitybarText,
      }}
      className="flex flex-col flex-grow gap-2 pt-2 pb-3 border-r-1 border-colorNeutralBorder"
    >
      <Button
        ariaLabel="Workspace Home"
        style={ButtonStyleOBj.normal}
        onPressStyle={ButtonStyleOBj.press}
        onHoverStyle={ButtonStyleOBj.hover}
        onPress={() => {
          if (wsName) {
            uiDispatch(nsmUI.closeSidebar());
            pageDispatch(
              goToWorkspaceHome({
                wsName,
              }),
            );
          }
        }}
        variant="transparent"
        tone="secondary"
        size="lg"
        leftIcon={
          <SingleCharIcon char={wsName?.[0]?.toLocaleUpperCase() || 'H'} />
        }
      />

      {sideBarComponents}
      <div className="flex-grow"></div>
      <Button
        ariaLabel="What's new"
        style={ButtonStyleOBj.normal}
        onPressStyle={ButtonStyleOBj.press}
        onHoverStyle={ButtonStyleOBj.hover}
        onPress={() => {
          uiDispatch(
            nsmUI.showDialog({
              dialogName: CHANGELOG_MODAL_NAME,
            }),
          );
        }}
        variant="transparent"
        tone="secondary"
        size="lg"
        leftIcon={<GiftIcon showDot={changelogHasUpdates} />}
      />
      <ActivitybarOptionsDropdown
        operationKeybindings={operationKeybindings}
        widescreen={widescreen}
      />
    </div>
  );
}
