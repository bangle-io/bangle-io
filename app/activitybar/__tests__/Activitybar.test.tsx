/**
 * @jest-environment @bangle.io/jsdom-env
 */
import { act, fireEvent, render, screen } from '@testing-library/react';
import React from 'react';

import type { SidebarType } from '@bangle.io/extension-registry';
import { Extension } from '@bangle.io/extension-registry';
import { changeSidebar } from '@bangle.io/slice-ui';
import type { TestInitialSliceStateOverride } from '@bangle.io/test-utils';
import { createBasicTestStore, TestStoreProvider } from '@bangle.io/test-utils';

import { Activitybar } from '../Activitybar';

jest.mock('@bangle.io/slice-ui', () => {
  const otherThings = jest.requireActual('@bangle.io/slice-ui');

  return {
    ...otherThings,
    changeSidebar: jest.fn(() => () => {}),
  };
});

let changeSidebarMock = changeSidebar as jest.MockedFunction<
  typeof changeSidebar
>;

const changeSidebarRet = jest.fn();
changeSidebarMock.mockImplementation(() => changeSidebarRet);

const createStoreWithSidebar = ({
  override,
  sidebar,
}: {
  override?: TestInitialSliceStateOverride;
  sidebar?: SidebarType;
} = {}) => {
  return createBasicTestStore({
    extensions: [
      Extension.create({
        name: 'test-ext-123',
        application: {
          sidebars: sidebar ? [sidebar] : undefined,
        },
      }),
    ],
    useEditorManagerSlice: true,
    useUISlice: true,
    overrideInitialSliceState: {
      ...override,
      uiSlice: {
        widescreen: true,
        ...override?.uiSlice,
      },
    },
  });
};

test('renders when no sidebars', () => {
  let { store } = createStoreWithSidebar();

  let result = render(
    <div>
      <TestStoreProvider bangleStore={store} bangleStoreChanged={0}>
        <Activitybar></Activitybar>
      </TestStoreProvider>
    </div>,
  );

  expect(result.container).toMatchSnapshot();
});

test('renders when there is sidebar', () => {
  let { store } = createStoreWithSidebar({
    sidebar: {
      name: 'sidebar::test-ext-123:sidebar-123',
      title: 'search notes',
      activitybarIcon: <span>test-search-icon</span>,
      ReactComponent: () => <p>search notes</p>,
      hint: 'test-search-hint',
    },
  });

  let result = render(
    <div>
      <TestStoreProvider bangleStore={store} bangleStoreChanged={0}>
        <Activitybar />
      </TestStoreProvider>
    </div>,
  );

  expect(result.container.innerHTML).toContain('test-search-icon');
  expect(result.container).toMatchSnapshot();
  expect(result.container.innerHTML).not.toContain('BU_is-active');
});

test('renders when sidebar is active', () => {
  let { store } = createStoreWithSidebar({
    override: {
      uiSlice: {
        sidebar: 'sidebar::test-ext-123:sidebar-123',
      },
    },
    sidebar: {
      name: 'sidebar::test-ext-123:sidebar-123',
      title: 'search notes',
      activitybarIcon: <span>test-search-icon</span>,
      ReactComponent: () => <p>search notes</p>,
      hint: 'test-search-hint',
    },
  });

  let result = render(
    <div>
      <TestStoreProvider bangleStore={store} bangleStoreChanged={0}>
        <Activitybar></Activitybar>
      </TestStoreProvider>
    </div>,
  );

  expect(result.container).toMatchSnapshot();
  expect(result.container.innerHTML).toContain('BU_is-active');
});

test('inactive sidebar is dispatched correctly', () => {
  let { store } = createStoreWithSidebar({
    override: {
      uiSlice: {
        sidebar: undefined,
      },
    },
    sidebar: {
      name: 'sidebar::test-ext-123:sidebar-123',
      title: 'search notes',
      activitybarIcon: <span>test-search-icon</span>,
      ReactComponent: () => <p>search notes</p>,
      hint: 'test-search-hint',
    },
  });

  let result = render(
    <div>
      <TestStoreProvider bangleStore={store} bangleStoreChanged={0}>
        <Activitybar></Activitybar>
      </TestStoreProvider>
    </div>,
  );
  act(() => {
    fireEvent.click(result.getByRole('button', { name: 'test-search-hint' }));
  });

  expect(changeSidebarMock).toBeCalledTimes(1);
  expect(changeSidebarMock).nthCalledWith(
    1,
    'sidebar::test-ext-123:sidebar-123',
  );
});

test('active sidebar is toggled off correctly', () => {
  let { store } = createStoreWithSidebar({
    override: {
      uiSlice: {
        sidebar: 'sidebar::test-ext-123:sidebar-123',
      },
    },
    sidebar: {
      name: 'sidebar::test-ext-123:sidebar-123',
      title: 'search notes',
      activitybarIcon: <span>test-search-icon</span>,
      ReactComponent: () => <p>search notes</p>,
      hint: 'test-search-hint',
    },
  });

  let result = render(
    <div>
      <TestStoreProvider bangleStore={store} bangleStoreChanged={0}>
        <Activitybar></Activitybar>
      </TestStoreProvider>
    </div>,
  );
  act(() => {
    fireEvent.click(result.getByRole('button', { name: 'test-search-hint' }));
  });

  expect(changeSidebarMock).toBeCalledTimes(1);
  expect(changeSidebarMock).nthCalledWith(
    1,
    'sidebar::test-ext-123:sidebar-123',
  );
});

test('activitybarIconShow is respected', async () => {
  let activitybarIconShow = jest.fn(() => false);

  let { store } = createStoreWithSidebar({
    override: {
      uiSlice: {
        sidebar: 'sidebar::test-ext-123:sidebar-123',
      },
    },
    sidebar: {
      name: 'sidebar::test-ext-123:sidebar-123',
      title: 'search notes',
      activitybarIcon: <span>test-search-icon</span>,
      ReactComponent: () => <p>search notes</p>,
      hint: 'test-search-hint',
      activitybarIconShow,
    },
  });
  let { rerender } = render(
    <div>
      <TestStoreProvider bangleStore={store} bangleStoreChanged={0}>
        <Activitybar></Activitybar>
      </TestStoreProvider>
    </div>,
  );

  expect(screen.queryAllByLabelText('test-search-hint')).toEqual([]);

  activitybarIconShow.mockImplementation(() => true);

  let { store: newStore } = createStoreWithSidebar({
    override: {
      uiSlice: {
        sidebar: 'sidebar::test-ext-123:sidebar-123',
      },
    },
    sidebar: {
      name: 'sidebar::test-ext-123:sidebar-123',
      title: 'search notes',
      activitybarIcon: <span>test-search-icon</span>,
      ReactComponent: () => <p>search notes</p>,
      hint: 'test-search-hint',
      activitybarIconShow,
    },
  });

  rerender(
    <div>
      <TestStoreProvider bangleStore={newStore} bangleStoreChanged={1}>
        <Activitybar></Activitybar>
      </TestStoreProvider>
    </div>,
  );

  expect(screen.queryAllByLabelText('test-search-hint')).toMatchInlineSnapshot(`
    [
      <button
        aria-label="test-search-hint"
        class="border-l-2 BU_is-active  text-lg font-600 h-11 min-w-12 px-4  select-none inline-flex justify-center items-center rounded-md whitespace-nowrap overflow-hidden py-1 transition-all duration-100 cursor-pointer "
        style="background-color: transparent; border-radius: 0; padding: 0px;"
        type="button"
      >
        <span
          class="flex flex-grow-1 overflow-hidden "
          style="justify-content: center;"
        >
          <span>
            <span
              class="w-7 h-7  "
            >
              test-search-icon
            </span>
          </span>
        </span>
      </button>,
    ]
  `);
});
