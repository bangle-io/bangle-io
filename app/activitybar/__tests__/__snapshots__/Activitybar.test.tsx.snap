// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when no sidebars 1`] = `
<div>
  <div>
    <div
      class="flex flex-col flex-grow gap-2 pt-2 pb-3 border-r-1 border-colorNeutralBorder"
      data-testid="app-activitybar_activitybar"
    >
      <button
        aria-label="Workspace Home"
        class="text-lg font-600 h-11 min-w-12 px-4  select-none inline-flex justify-center items-center rounded-md whitespace-nowrap overflow-hidden py-1 transition-all duration-100 cursor-pointer "
        style="background-color: transparent; border-radius: 0; padding: 0px;"
        type="button"
      >
        <span
          class="flex flex-grow-1 overflow-hidden "
          style="justify-content: center;"
        >
          <span>
            <svg
              class="w-7 h-7  "
              fill="currentColor"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="11"
                stroke-width="2"
                style="fill: none;"
              />
              <text
                alignment-baseline="central"
                stroke-width="0"
                text-anchor="middle"
                x="50%"
                y="50%"
              >
                H
              </text>
            </svg>
          </span>
        </span>
      </button>
      <div
        class="flex-grow"
      />
      <button
        aria-label="What's new"
        class="text-lg font-600 h-11 min-w-12 px-4  select-none inline-flex justify-center items-center rounded-md whitespace-nowrap overflow-hidden py-1 transition-all duration-100 cursor-pointer "
        style="background-color: transparent; border-radius: 0; padding: 0px;"
        type="button"
      >
        <span
          class="flex flex-grow-1 overflow-hidden "
          style="justify-content: center;"
        >
          <span>
            <svg
              class="w-7 h-7  "
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 8v13m0-13V6a2 2 0 112 2h-2zm0 0V5.5A2.5 2.5 0 109.5 8H12zm-7 4h14M5 12a2 2 0 110-4h14a2 2 0 110 4M5 12v7a2 2 0 002 2h10a2 2 0 002-2v-7"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </span>
        </span>
      </button>
      <button
        aria-label="options menu"
        aria-pressed="false"
        class="text-lg font-600 h-11 min-w-12 px-4  select-none inline-flex justify-center items-center rounded-md whitespace-nowrap overflow-hidden py-1 transition-all duration-100 cursor-pointer "
        style="background-color: transparent; border-radius: 0; padding: 0px;"
        type="button"
      >
        <span
          class="flex flex-grow-1 overflow-hidden "
          style="justify-content: center;"
        >
          <span>
            <svg
              class="w-7 h-7  "
              fill="currentColor"
              viewBox="0 0 18 18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="#ff13dc"
                height="18"
                opacity="0"
                width="18"
              />
              <path
                d="M16.45,7.8965H14.8945a5.97644,5.97644,0,0,0-.921-2.2535L15.076,4.54a.55.55,0,0,0,.00219-.77781L15.076,3.76l-.8365-.836a.55.55,0,0,0-.77781-.00219L13.4595,2.924,12.357,4.0265a5.96235,5.96235,0,0,0-2.2535-.9205V1.55a.55.55,0,0,0-.55-.55H8.45a.55.55,0,0,0-.55.55V3.106a5.96235,5.96235,0,0,0-2.2535.9205l-1.1-1.1025a.55.55,0,0,0-.77781-.00219L3.7665,2.924,2.924,3.76a.55.55,0,0,0-.00219.77781L2.924,4.54,4.0265,5.643a5.97644,5.97644,0,0,0-.921,2.2535H1.55a.55.55,0,0,0-.55.55V9.55a.55.55,0,0,0,.55.55H3.1055a5.967,5.967,0,0,0,.921,2.2535L2.924,13.4595a.55.55,0,0,0-.00219.77782l.00219.00218.8365.8365a.55.55,0,0,0,.77781.00219L4.5405,15.076,5.643,13.9735a5.96235,5.96235,0,0,0,2.2535.9205V16.45a.55.55,0,0,0,.55.55H9.55a.55.55,0,0,0,.55-.55V14.894a5.96235,5.96235,0,0,0,2.2535-.9205L13.456,15.076a.55.55,0,0,0,.77782.00219L14.236,15.076l.8365-.8365a.55.55,0,0,0,.00219-.77781l-.00219-.00219L13.97,12.357a5.967,5.967,0,0,0,.921-2.2535H16.45a.55.55,0,0,0,.55-.55V8.45a.55.55,0,0,0-.54649-.55349ZM11.207,9A2.207,2.207,0,1,1,9,6.793H9A2.207,2.207,0,0,1,11.207,9Z"
              />
            </svg>
          </span>
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`renders when sidebar is active 1`] = `
<div>
  <div>
    <div
      class="flex flex-col flex-grow gap-2 pt-2 pb-3 border-r-1 border-colorNeutralBorder"
      data-testid="app-activitybar_activitybar"
    >
      <button
        aria-label="Workspace Home"
        class="text-lg font-600 h-11 min-w-12 px-4  select-none inline-flex justify-center items-center rounded-md whitespace-nowrap overflow-hidden py-1 transition-all duration-100 cursor-pointer "
        style="background-color: transparent; border-radius: 0; padding: 0px;"
        type="button"
      >
        <span
          class="flex flex-grow-1 overflow-hidden "
          style="justify-content: center;"
        >
          <span>
            <svg
              class="w-7 h-7  "
              fill="currentColor"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="11"
                stroke-width="2"
                style="fill: none;"
              />
              <text
                alignment-baseline="central"
                stroke-width="0"
                text-anchor="middle"
                x="50%"
                y="50%"
              >
                H
              </text>
            </svg>
          </span>
        </span>
      </button>
      <button
        aria-label="test-search-hint"
        class="border-l-2 BU_is-active  text-lg font-600 h-11 min-w-12 px-4  select-none inline-flex justify-center items-center rounded-md whitespace-nowrap overflow-hidden py-1 transition-all duration-100 cursor-pointer "
        style="background-color: transparent; border-radius: 0; padding: 0px;"
        type="button"
      >
        <span
          class="flex flex-grow-1 overflow-hidden "
          style="justify-content: center;"
        >
          <span>
            <span
              class="w-7 h-7  "
            >
              test-search-icon
            </span>
          </span>
        </span>
      </button>
      <div
        class="flex-grow"
      />
      <button
        aria-label="What's new"
        class="text-lg font-600 h-11 min-w-12 px-4  select-none inline-flex justify-center items-center rounded-md whitespace-nowrap overflow-hidden py-1 transition-all duration-100 cursor-pointer "
        style="background-color: transparent; border-radius: 0; padding: 0px;"
        type="button"
      >
        <span
          class="flex flex-grow-1 overflow-hidden "
          style="justify-content: center;"
        >
          <span>
            <svg
              class="w-7 h-7  "
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 8v13m0-13V6a2 2 0 112 2h-2zm0 0V5.5A2.5 2.5 0 109.5 8H12zm-7 4h14M5 12a2 2 0 110-4h14a2 2 0 110 4M5 12v7a2 2 0 002 2h10a2 2 0 002-2v-7"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </span>
        </span>
      </button>
      <button
        aria-label="options menu"
        aria-pressed="false"
        class="text-lg font-600 h-11 min-w-12 px-4  select-none inline-flex justify-center items-center rounded-md whitespace-nowrap overflow-hidden py-1 transition-all duration-100 cursor-pointer "
        style="background-color: transparent; border-radius: 0; padding: 0px;"
        type="button"
      >
        <span
          class="flex flex-grow-1 overflow-hidden "
          style="justify-content: center;"
        >
          <span>
            <svg
              class="w-7 h-7  "
              fill="currentColor"
              viewBox="0 0 18 18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="#ff13dc"
                height="18"
                opacity="0"
                width="18"
              />
              <path
                d="M16.45,7.8965H14.8945a5.97644,5.97644,0,0,0-.921-2.2535L15.076,4.54a.55.55,0,0,0,.00219-.77781L15.076,3.76l-.8365-.836a.55.55,0,0,0-.77781-.00219L13.4595,2.924,12.357,4.0265a5.96235,5.96235,0,0,0-2.2535-.9205V1.55a.55.55,0,0,0-.55-.55H8.45a.55.55,0,0,0-.55.55V3.106a5.96235,5.96235,0,0,0-2.2535.9205l-1.1-1.1025a.55.55,0,0,0-.77781-.00219L3.7665,2.924,2.924,3.76a.55.55,0,0,0-.00219.77781L2.924,4.54,4.0265,5.643a5.97644,5.97644,0,0,0-.921,2.2535H1.55a.55.55,0,0,0-.55.55V9.55a.55.55,0,0,0,.55.55H3.1055a5.967,5.967,0,0,0,.921,2.2535L2.924,13.4595a.55.55,0,0,0-.00219.77782l.00219.00218.8365.8365a.55.55,0,0,0,.77781.00219L4.5405,15.076,5.643,13.9735a5.96235,5.96235,0,0,0,2.2535.9205V16.45a.55.55,0,0,0,.55.55H9.55a.55.55,0,0,0,.55-.55V14.894a5.96235,5.96235,0,0,0,2.2535-.9205L13.456,15.076a.55.55,0,0,0,.77782.00219L14.236,15.076l.8365-.8365a.55.55,0,0,0,.00219-.77781l-.00219-.00219L13.97,12.357a5.967,5.967,0,0,0,.921-2.2535H16.45a.55.55,0,0,0,.55-.55V8.45a.55.55,0,0,0-.54649-.55349ZM11.207,9A2.207,2.207,0,1,1,9,6.793H9A2.207,2.207,0,0,1,11.207,9Z"
              />
            </svg>
          </span>
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`renders when there is sidebar 1`] = `
<div>
  <div>
    <div
      class="flex flex-col flex-grow gap-2 pt-2 pb-3 border-r-1 border-colorNeutralBorder"
      data-testid="app-activitybar_activitybar"
    >
      <button
        aria-label="Workspace Home"
        class="text-lg font-600 h-11 min-w-12 px-4  select-none inline-flex justify-center items-center rounded-md whitespace-nowrap overflow-hidden py-1 transition-all duration-100 cursor-pointer "
        style="background-color: transparent; border-radius: 0; padding: 0px;"
        type="button"
      >
        <span
          class="flex flex-grow-1 overflow-hidden "
          style="justify-content: center;"
        >
          <span>
            <svg
              class="w-7 h-7  "
              fill="currentColor"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="11"
                stroke-width="2"
                style="fill: none;"
              />
              <text
                alignment-baseline="central"
                stroke-width="0"
                text-anchor="middle"
                x="50%"
                y="50%"
              >
                H
              </text>
            </svg>
          </span>
        </span>
      </button>
      <button
        aria-label="test-search-hint"
        class="border-l-2  text-lg font-600 h-11 min-w-12 px-4  select-none inline-flex justify-center items-center rounded-md whitespace-nowrap overflow-hidden py-1 transition-all duration-100 cursor-pointer "
        style="background-color: transparent; border-radius: 0; padding: 0px;"
        type="button"
      >
        <span
          class="flex flex-grow-1 overflow-hidden "
          style="justify-content: center;"
        >
          <span>
            <span
              class="w-7 h-7  "
            >
              test-search-icon
            </span>
          </span>
        </span>
      </button>
      <div
        class="flex-grow"
      />
      <button
        aria-label="What's new"
        class="text-lg font-600 h-11 min-w-12 px-4  select-none inline-flex justify-center items-center rounded-md whitespace-nowrap overflow-hidden py-1 transition-all duration-100 cursor-pointer "
        style="background-color: transparent; border-radius: 0; padding: 0px;"
        type="button"
      >
        <span
          class="flex flex-grow-1 overflow-hidden "
          style="justify-content: center;"
        >
          <span>
            <svg
              class="w-7 h-7  "
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 8v13m0-13V6a2 2 0 112 2h-2zm0 0V5.5A2.5 2.5 0 109.5 8H12zm-7 4h14M5 12a2 2 0 110-4h14a2 2 0 110 4M5 12v7a2 2 0 002 2h10a2 2 0 002-2v-7"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </span>
        </span>
      </button>
      <button
        aria-label="options menu"
        aria-pressed="false"
        class="text-lg font-600 h-11 min-w-12 px-4  select-none inline-flex justify-center items-center rounded-md whitespace-nowrap overflow-hidden py-1 transition-all duration-100 cursor-pointer "
        style="background-color: transparent; border-radius: 0; padding: 0px;"
        type="button"
      >
        <span
          class="flex flex-grow-1 overflow-hidden "
          style="justify-content: center;"
        >
          <span>
            <svg
              class="w-7 h-7  "
              fill="currentColor"
              viewBox="0 0 18 18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="#ff13dc"
                height="18"
                opacity="0"
                width="18"
              />
              <path
                d="M16.45,7.8965H14.8945a5.97644,5.97644,0,0,0-.921-2.2535L15.076,4.54a.55.55,0,0,0,.00219-.77781L15.076,3.76l-.8365-.836a.55.55,0,0,0-.77781-.00219L13.4595,2.924,12.357,4.0265a5.96235,5.96235,0,0,0-2.2535-.9205V1.55a.55.55,0,0,0-.55-.55H8.45a.55.55,0,0,0-.55.55V3.106a5.96235,5.96235,0,0,0-2.2535.9205l-1.1-1.1025a.55.55,0,0,0-.77781-.00219L3.7665,2.924,2.924,3.76a.55.55,0,0,0-.00219.77781L2.924,4.54,4.0265,5.643a5.97644,5.97644,0,0,0-.921,2.2535H1.55a.55.55,0,0,0-.55.55V9.55a.55.55,0,0,0,.55.55H3.1055a5.967,5.967,0,0,0,.921,2.2535L2.924,13.4595a.55.55,0,0,0-.00219.77782l.00219.00218.8365.8365a.55.55,0,0,0,.77781.00219L4.5405,15.076,5.643,13.9735a5.96235,5.96235,0,0,0,2.2535.9205V16.45a.55.55,0,0,0,.55.55H9.55a.55.55,0,0,0,.55-.55V14.894a5.96235,5.96235,0,0,0,2.2535-.9205L13.456,15.076a.55.55,0,0,0,.77782.00219L14.236,15.076l.8365-.8365a.55.55,0,0,0,.00219-.77781l-.00219-.00219L13.97,12.357a5.967,5.967,0,0,0,.921-2.2535H16.45a.55.55,0,0,0,.55-.55V8.45a.55.55,0,0,0-.54649-.55349ZM11.207,9A2.207,2.207,0,1,1,9,6.793H9A2.207,2.207,0,0,1,11.207,9Z"
              />
            </svg>
          </span>
        </span>
      </button>
    </div>
  </div>
</div>
`;
